### DISCUSSION TITLE
v3 (3.0.6) vs v2 behavior regarding 'undefined' properties

### DISCUSSION THREAD

**Discussion by personalsoft-fabianobonin** [alpinejs/alpine#1496] [7. Bugs (archived)] [answered]
Given this html:

```html
    <div x-data="{ entity: {} }">
      <input type="text" :value="entity.field1">
    </div>
```

v2 shows a blank input

v3 shows a input filled with 'undefined'

In my application, when I am creating a new record, I usually create an empty entity { } inside data object and bind its properties to its respective inputs using :value and @change . In v3, all inputs are coming with the 'undefined' value.

Is it expected?

**Comment by ryangjchandler** [comment, marked as answer]
@personalsoft-fabianobonin This isn't an Alpine thing, more of a JS thing. Doing `entity.nonExistentField` will always return `undefined`.

In v2, I believe we were doing a check to see if the value was actually undefined before binding. I'll have to confirm though.

**Reply by calebporzio** [reply]
Yeah, I recall something like that - I'd be open to a PR with a test for it

**Reply by ryangjchandler** [reply]
@calebporzio Already on it.
